keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+
+      * `pod.spec.resources.limits[memory/cpu] < pod.spec.containers[i].resources.limits[memory/cpu]` is invalid.
+
+        A single container’s limit should not be allowed to exceed the pod level limit.
+
+#### Scheduler Changes
+
+The scheduler logic needs to be updated to support pod-level requests and limits.
+
+* **Scenario 1: Only container-level resources are specified**
+
+  No changes, same as existing.
+
+* **Scenario 2: Only pod-level resources are specified**
+
+  Consider pod-level requests to find the suitable node for pod placement.
+
+* **Scenario 3: Both container-level and pod-level resources are specified**
+
+  Scheduler will prioritize pod-level requests as the sole indicator of the pod’s resources needs. It will ignore container level requests. This ensures that the pod 
+  is scheduled on a node that can fulfill the overall resource requirements of the pod, regardless of the individual container requests.
+
+
+#### QoS Changes
+
+The QoS definitions need to be updated to consider pod-level resources. To attach a QoS class to a pod, the  pod’s resource specifications are evaluated in order:
+
+1. The values for resources specified in the pod’s `resources` stanza.
+2. The value of resources specified in the container’s `resources` stanza.
+
+In simple words, if resources are specified at pod-level, the values in pod-level `resources stanza` are given a preference in determining the QoS class.
+
+* **Guaranteed QoS Criteria**
+
+    One of the following two conditions must apply:
+
+    * Pod level requests and limits are specified for each resource, and request is equal to limit for each resource. The container level resource specification 
+    (specified or not specified) is not taken into account for QoS class determination in this case.
+    * In absence of pod-level resources specifications, every container in the pod has requests and limits specified for each resource and all container requests
+     and limits are equal.
+
+* **Burstable QoS Criteria**
+
+    For Burstable QoS, the pod does not meet the criteria for Guaranteed QoS class and one of the following two conditions must apply:
+
+    * Pod level requests or limits is specified. If both pod-level request and limit are specified, the request should be less than the limit. The container level
+    resource specification (specified or not specified) is not taken into account for QoS class determination in this case.
+    * In absence of pod-level resources specifications, at least one container within the pod has requests or limits specified. If both request and limit are 
+    specified for a resource at container-level, the request should be less than the limit.
+
+* **BestEffort QoS Criteria**
+
+    None of pod-level or container-level resources requests or limits are specified.
+
+#### Init Containers
+
+With the current implementation (before this KEP), Pod’s effective resource request/limit is the higher of:
```

Comment:
please include the sidecars in this formulae. The calculation is more complicated these days

Reply:
+1 for expanding this to include sidecar containers